### DISCUSSION TITLE
SSR not working with graphql@16

### DISCUSSION THREAD

**Discussion by liuderchi** [apollographql/apollo-server#5911] [Q&A]
Hi, I have SSR issues when upgrading `graphql` from `v15` to `v16` with following GQL tech stack:

- `next@latest`
- `apollo-server-express@3.5`
-  `@apollo/client@^3.4.17`
- `apollo-datasource-rest@^3.4.0`
- `@graphql-tools/utils` -> used for custom GQL directive
- `@graphql-tools/schema` -> used `makeExecutableSchema` for SSR support with `SchemaLink`
  - e.g.

```js
import { ApolloServer } from 'apollo-server-express'
import { SchemaLink } from '@apollo/client/link/schema'
import { makeExecutableSchema } from '@graphql-tools/schema'

// ...
req.schemaLink = new SchemaLink({
  schema: makeExecutableSchema({
    typeDefs,
    resolvers,
  })
})
```

Observed that GQL query results are _not resolved at server-side-rendering phase_, leading to some issues since I hope to query sensitive data only at server side.

---

And observed one thing interesting when debugging:
- this issue still exists only after this PR included in `graphql@16`:
  - https://github.com/graphql/graphql-js/pull/2904, which is one of the [breaking changes from `v16`](https://github.com/graphql/graphql-js/releases/tag/v16.0.0)
  - (this was tested with locally built `graphql` from git commit tree)

So, I'm wondering do apollo server and `SchemaLink` from `@apollo/client` both handle this breaking change in `graphql@16`?

Really appreciate your attention for this issue 🙇

**Comment by brainkim** [comment]
@liuderchi Sorry to hear you’re having trouble! The first version of Apollo Client with GraphQL 16 support would be ^3.5. Can you try that? No promises about whether it will fix your problem.

If it doesn’t, can you tell us a little more about the error that you’re seeing? Like any stack traces you have would be helpful in identifying the problem. A reproducible example would be even more helpful!
